### ISSUE TITLE
Establish systems for processing leads

### ISSUE THREAD

**Issue by colliand** [2i2c-org/team-compass#325]
### Description

2i2c receives expressions of interest in service/collaboration through the contact form on our web site. 2i2c team members receive incidental expressions of interest when we give talks, interact with people at conferences, talk to colleagues, etc. 

2i2c should establish solid systems to process these leads. We need to reliably acknowledge these messages, follow up with sales or technical engagement, etc. 

The current approach to processing these leads relies upon @choldgraf doing the work in between his other tasks. The Sustainability Team should consider this issue and establish improved methods for converting interest in 2i2c into a source of revenue. 

### Value / benefit

There is a lot of value for 2i2c to establishing an effective system to process leads. Some of the leads we receive are from motivated people with an interest in open source. They sometimes engage because they find strong alignment with 2i2c's values. These kinds of people often recognize a problem and are working on solutions. Slow response from 2i2c may lead some of these people to find their own paths toward solutions that go against 2i2c's long-term goals. 

### Implementation details

Leads need propagate reliably into a CRM so that a sales person can evaluate and then take action. 2i2c has a proto-CRM that consists of a mixture of AirTable and Google docs. Work is needed here.

**Comment by choldgraf** [comment]
There was some conversation about this in the CS&S Slack and another project lead put together a short write-up of CRMs. Sharing here in case it's useful:

- https://docs.google.com/document/d/1fiZkwqUdWFHRmjvorX0Wespifogcxxme24B9u-Blr_8/edit

**Comment by jmunroe** [comment]
Re: Freshworks.   We've been assigned 

Siddharth Banga
Senior Account Manager, Freshworks
siddharth.banga@freshworks.com

to support our 2i2c account.  I've just taken a 30 minute intro meeting with him to share more information about 2i2c, our needs, and to learn about what else Freshworks may be be able to provide to us.

I am still learning how tightly integrated  their CRM product 'Freshsales' is with 'Freshdesk'.  I am not anticipating any change in the near future but I see a possible path forward where we reduce the number of 'FreshDesk' seats down to 6 for our engineers, and shift over to 4 seats under FreshSales (same price /user/month for either product).  If the two Freshworks products are indeed well integrated it is possible that would our Community-Partnerships team access to CRM features while still giving visibility to the content of all support tickets across all of 2i2c for the same current cost we are currently paying.

None of this precludes us researching other CRMs.  @colliand  -- we can put this on the back burner to pick up at some future point.

**Comment by choldgraf** [comment]
### Prototype of an AirTable-based system

In a recent strategy meeting we discussed recent improvements to our contracts / invoicing infrastructure using CS&S's AirTables:

- #545 / #704 

Based on those conversations we decided to try building a system for leads that also uses AirTable, so that we can integrate it with the "source of truth" for contracts coming from CS&S. There were a few considerations for this:

- We want engineering to have visibility into this process, so **internal team discussion needs to use GitHub**
- We want external partners to have visibility into our conversations etc, **external discussion and planning needs to use Google Drive and docs**
- We want to manage this system with minimal manual actions and toil, so we should try to make AirTable a single "control point" that doesn't require many manual actions elsewhere.
- We want this system to integrate with our Contracts table so we should define a process that hands off a "lead" into a `Contract` record

### Links

- [AirTable prototype of leads table](https://airtable.com/appbjBTRIbgRiElkr/tblmRU6U53i8o7z2I/viwVddEUxZE5WHXFB?blocks=hide)
- [Video of overview of the AirTable prototype](https://2i2c.slack.com/files/UKUUCELKX/F053X20F9FH/screenshare_-_2023-04-21_11_27_20_am.webm)

**Comment by choldgraf** [comment]
### Google Drive folders can sync to airtable

Another thing I just discovered. Apparently Google Drive and AirTable have an auto-sync option, so we can create an AirTable table that mirrors the structure of a google drive folder.

For example, [here's a table that mirrors the `Agreements` folder](https://airtable.com/appbjBTRIbgRiElkr/tbl7xHdL9wTxR4lwJ/viwdKhRAI8ckuvHNB?blocks=hide):

![image](https://user-images.githubusercontent.com/1839645/234261797-855db62c-80cf-4ac6-a30c-3e1630a00c8a.png)

**Comment by choldgraf** [comment]
### Agreement to use AirTable for this

In the [last organization / strategy meeting](https://docs.google.com/document/d/1HoNX8T8IQ1uhS2ryi1r9iS-nSbPT1b1Y7HsjosbHme8/edit#heading=h.a5rpmstzzo7c) @choldgraf gave a quick demo of the AirTable setup and @colliand + @jmunroe agreed that AirTable should be used as a source of truth for tracking leads. We don't know the exact setup so this is the next step.

We didn't know what the order of database operations should be. For example, should we first create a github issue, then an airtable record, and then link it to a Google Drive folder we create? Should we do this automatically? And how could we convert over the current systems of leads into this new dashboard?

**Next step**: @colliand will experiment and set up a minimal AirTable that works for him

### Proposal: use a synced AirTable from either Google Drive or from GitHub Issues to define leads

I propose that, at least to begin, we treat either **GitHub Issues in `leads/`** or **Folders in the `Agreements/` folder** as the source of truth for "which leads are currently active". if we did so, then we could use an AirTable "synced view" to GitHub / Google Drive to define all of our leads, the "Name" would be synced to the issue title or the folder title, and we'd then only need to manually link one extra thing rather than two.

**Comment by choldgraf** [comment]
This process now exists and I think the only remaining thing is to document the process that @colliand is currently using. Is that right @colliand ?

**Comment by colliand** [comment]
Yes. We have a single source of truth for managing leads. The associated systems need to be documented.
